### PR TITLE
Sharing: Add Notice for Atomic Private Sites

### LINKED ISSUE #54299: Sharing: Private Atomic Site (Can we update the wordings?)
Hey team! 

I'd figure I can open GH issue here and we can transfer it to somewhere else if needed/if this is not the right realm 😄 

Sharing buttons will not be available/not accessible when an Atomic website is set to Private -- this is [a known circumstance that we know of](https://github.com/Automattic/wp-calypso/issues/43988).

The problem is with the wordings; although I can see there are two possibilities:

- Site is public, but Sharing module on Jetpack is not enabled, or...
- Site is private, so Sharing module is not available

I'm referring to this:

Screenshot 2021-07-06 at 9 35 06 AM

The sentence "Adding sharing buttons need the Sharing Buttons module from Jetpack to be enabled" might not be super clear. 

When an AT site is set to Private, HEs won't be able to see Sharing module on Blog RC and CLI. This can cause confusion and might take quite some time to troubleshoot/gut check, while the actual issue is site privacy settings that can be toggled on and off by the user themselves.

I propose we can change the wordings to:

```
Sharing feature is not available right now. Please check:
- If the website is set to Public from Settings > General
- If Sharing Buttons module is active from Jetpack > Settings > Sharing
```

Is it possible/doable? I can see if the notification modal itself can take quite amount of space, but maybe we can provide better wordings so the user can do their first initial troubleshooting on their own.

Thank youuu!

### Context
HE discussion and personal test

#### Does this happen on simple or atomic sites or both?
Atomic sites

#### Level of impact (Does it block purchases? Does it affect more than just one site?)
More than one site, and not blocking any purchase

#### Reproducibility (Consistent, Intermittent) Leave empty for consistent.
Consistent

### PR DESCRIPTION
**Pull request by Aurorum**
#### Changes proposed in this Pull Request

* Adds a check for Atomic sites which are set to Private, for which Sharing is disabled

#### Testing instructions

When a site is marked as Private when Atomic, verify that the appropriate notice appears at `/marketing/sharing-buttons/url`.

Screenshot 2021-07-06 at 18 33 52

cc @mmtr 

Fixes #54299

### CODE REVIEW

File: client/my-sites/marketing/buttons/buttons.jsx [hunk 1]
```diff
@@ -159,12 +162,19 @@ class SharingButtons extends Component {
 					<Notice
 						status="is-warning"
 						showDismiss={ false }
-						text={ translate(
-							'Adding sharing buttons need the Sharing Buttons module from Jetpack to be enabled'
-						) }
+						text={
+							isPrivate
+								? translate( 'Adding sharing buttons requires your site to be marked as Public.' )
```
Comment (a8ci18n) [inline client/my-sites/marketing/buttons/buttons.jsx:167]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Review by mmtr — APPROVED** [review]
Works great! Thanks for your contribution, @Aurorum.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6152196

Thank you @Aurorum for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6152196).

**Pull request merged**